compiler/z/codegen/UnaryEvaluator.cpp

```diff
@@ -520,9 +520,14 @@ OMR::Z::TreeEvaluator::l2aEvaluator(TR::Node * node, TR::CodeGenerator * cg)
    node->setRegister(source);
 
 
-   //printf("Node %p is first child\n", firstChild); fflush(stdout);
-   //traceMsg(comp, "Node %p is first child\n", firstChild);
-   cg->decReferenceCount(firstChild);
+   if (shouldGenerateGuardedLoadsForCompressedLoad || firstChild->containsCompressionSequence())
```

Comment:
I don't think this is correct. If concurrent scavenge is not enabled then we wouldn't be anchoring the `l2a` nodes to treetops, hence `shouldGenerateGuardedLoadsForCompressedLoad` will be `false`. This means you will skip evaluation of `l2a` nodes when concurrent scavenge is not enabled which is incorrect. Please run a personal build with this change on upstream projects to see what I mean.

As a general rule every node you evaluate within an evaluator you should decrement its reference count. Any register you allocate that is not attached to any node you need to call `stopUsingRegister` on it. The fix should look something like this:

```
if (shouldGenerateGuardedLoadsForCompressedLoad && firstChild->containsCompressionSequence())
      {
      cg->decReferenceCount(firstChild->getFirstChild());
      }
   else
      {
      cg->decReferenceCount(firstChild);
      }
```

However I don't think this is the final solution. As @0xdaryl who decrements the reference count of `firstChild`? I don't think this is properly handled at the moment. Please run your use case with `-Xjit:paranoidOptCheck,enableParanoidRefCountChecks` and I suspect you will see asserts regarding reference counts of nodes not being zero at the end of the compilation.